### PR TITLE
feat: make changes to PresenceData typings and docs

### PR DESCRIPTION
**Pull request by iShibi**
**Please describe the changes this PR makes and why it should be merged:**

Fixes: #5316 
Also, for some reason, jsdoc description of `ActivityOptions` was missing the `name` property.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by iShibi** [comment]
Just for context purpose: I got confused and thought `shardID` goes inside `activities` in `PresenceData` just like it does in `ActivityOptions`. Fixed my mistake in 7179689 😓

File: src/structures/ClientUser.js [hunk 1]
```diff
@@ -95,10 +95,10 @@ class ClientUser extends Structures.get('User') {
...
    * @property {PresenceStatusData} [status] Status of the user
    * @property {boolean} [afk] Whether the user is AFK
-   * @property {Object} [activity] Activity the user is playing
-   * @property {string} [activity.name] Name of the activity
-   * @property {ActivityType|number} [activity.type] Type of the activity
-   * @property {string} [activity.url] Twitch / YouTube stream URL
+   * @property {Object[]} [activities] Activity the user is playing
+   * @property {string} [activities[].name] Name of the activity
+   * @property {ActivityType|number} [activities[].type] Type of the activity
+   * @property {string} [activities[].url] Twitch / YouTube stream URL
```
Comment (NotSugden) [inline src/structures/ClientUser.js:101]: have you tested if this works with the docs? if not perhaps make a type for `activities`

**Review by NotSugden — CHANGES_REQUESTED** [review]

File: src/structures/ClientUser.js (hunk 1 above)
Comment (iShibi) [reply src/structures/ClientUser.js:101]: This is how it looks in the docs:


Normal jsdoc for comparison:


How about we make a new `ActivitiesOptions` type, with `name`, `type`, and `url` property. Then we can document this `PresenceData#activities` to be an array of `ActivitiesOptions`.

Comment (NotSugden) [reply src/structures/ClientUser.js:101]: yeah that would be best, because `activities[].name` looks kinda funky

File: typings/index.d.ts [hunk 2]
```diff
@@ -2124,6 +2124,12 @@ declare module 'discord.js' {
 
   type ActivityFlagsString = 'INSTANCE' | 'JOIN' | 'SPECTATE' | 'JOIN_REQUEST' | 'SYNC' | 'PLAY';
 
+  interface ActivitiesOptions {
+    name?: string;
+    url?: string;
+    type?: ActivityType | number;
+  }
```
Comment (NotSugden) [inline typings/index.d.ts:2131]:
```suggestion
  type ActivitiesOptions = Omit<ActivityOptions, 'shardID'>;
```

**Review by MattIPv4 — APPROVED** [review]
Lgtm

**Review by wolf-yuan-6115 — APPROVED** [review]

**Review by Nimplex — APPROVED** [review]

**Review by wolf-yuan-6115 — APPROVED** [review]
Approved

**Comment by kyranet** [comment]
This needs its conflicts resolved.

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**